### PR TITLE
Prompt user to create account on uwflow.com if new account

### PR DESCRIPTION
**Pull request by divad12**
If a user tries to log in with Facebook but doesn't already have an account on uwflow.com, we'll show this:

image

Ideally, we would create an account directly on the mobile app, but this involves more work... we'll have to send up basic Facebook info and friends list on every uwflow.com auth request and involves some more API work on the server.

There's also the problem of populating user account data... ideally the user would just do this on desktop, and we don't want to rebuild that flow (of pasting in transcript + schedule) on mobile.

### CODE REVIEW

File: src/com/uwflow/flow_android/LoginActivity.java [hunk 1]
```diff
@@ -49,13 +53,27 @@ public void onSuccess(JSONObject response) {
...
+                                        // Clear Facebook session
+                                        if (Session.getActiveSession() != null) {
+                                            Session.getActiveSession().closeAndClearTokenInformation();
+                                        }
+                                        Session.setActiveSession(null);
+
+                                        // Remove any cookies
+                                        FlowAsyncClient.clearCookie();
+
+                                        if (error.toLowerCase().contains("create an account")) {
```
Comment (mduan) [inline src/com/uwflow/flow_android/LoginActivity.java:72]: I don't have enough context for this change, but it seems strange that we're doing string comparison to determine whether the user needs to create an account.

**Comment by mduan** [comment]
UI-wise, lGTM

File: src/com/uwflow/flow_android/LoginActivity.java (hunk 1 above)
Comment (divad12) [inline src/com/uwflow/flow_android/LoginActivity.java:72]: I agree this is not ideal. This is due to the error that's returned from the server:

``` python
    if not user:
        raise api_util.ApiForbiddenError('No user with fbid %s exists. '
                'Create an account at uwflow.com.' % fbid)
```

From https://github.com/UWFlow/rmc/blob/master/server/api/v1.py#L248

(There are other 403 errors returned from that endpoint.)

What do you think we should do? Return an error code from the API?

Comment (mduan) [inline src/com/uwflow/flow_android/LoginActivity.java:72]: Yeah, I think an error code would be better. We could still return an HTTP 403 error, but also return an application error code that indicates the account does not exist. But for now, I think this is fine, but you might want to add a TODO.

**Comment by divad12** [comment]
Piggy-backing on this PR, instead of showing a progress modal, show loading spinner directly on login screen when logging in (as advised by http://developer.android.com/guide/topics/ui/dialogs.html).

image

**Comment by mduan** [comment]
What will happen when the back button is pressed when the create account modal is shown? Would the user be taken back to the login screen with the spinning spinner?

**Comment by divad12** [comment]
A user can't go back to this login page after they've logged in successfully — we `finish()` the `LoginActivity`.

**Comment by mduan** [comment]
I mean for the case the user doesn't have an account and the "create an account" modal shows up. What happens if they click back?

**Comment by divad12** [comment]
That'll just dismiss the dialog and bring them to the login screen showing the initial state, which is fine.

**Comment by mduan** [comment]
Okay, that was the only case I wasn't clear about.

LGTM.

**Comment by jasperfung** [comment]
Yesssssssssssssssss thanks for the integrated loading spinner. Long time coming. 

LGTM. Go forth and merge.

**Pull request merged**